File: packages/core/src/engine/verbs/rename.ts
/*!
 * Copyright (c) Microsoft. All rights reserved.
 * Licensed under the MIT license. See LICENSE file in the project.
 */
import type ColumnTable from 'arquero/dist/types/table/column-table'
import type { TableStore } from '../../index.js'
import type { RenameArgs, Step } from '../../types.js'

/**
 * Executes an arquero column rename.
 * @param step
 * @param store
 * @returns
 */
export async function rename(
	step: Step,
	store: TableStore,
): Promise<ColumnTable> {
	const { input, args } = step
	const { columns } = args as RenameArgs
	const inputTable = await store.table(input)
	return inputTable.rename(columns)
}

File: python/data_wrangling_components/engine/verbs/rename.py
#
# Copyright (c) Microsoft. All rights reserved.
# Licensed under the MIT license. See LICENSE file in the project.
#

from typing import Dict

from dataclasses import dataclass

from data_wrangling_components.table_store import TableStore
from data_wrangling_components.types import Step


@dataclass
class RenameArgs:
    columns: Dict[str, str]


def rename(step: Step, store: TableStore):
    """Rename columns in a table.

    :param step:
        Parameters to execute the operation.
        See :py:class:`~data_wrangling_components.engine.verbs.orderby.OrderByArgs`.
    :type step: Step
    :param store:
        Table store that contains the inputs to be used in the execution.
    :type store: TableStore

    :return: new table with the result of the operation.
    """
    args = RenameArgs(columns=step.args["columns"])
    input_table = store.get(step.input)
    return input_table.rename(columns=args.columns)
